lib/features/profiles/widgets/profile_image_deleter.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:flutter/material.dart';
+import 'package:gruene_app/app/theme/theme.dart';
+import 'package:gruene_app/features/profiles/domain/profiles_api_service.dart';
+import 'package:gruene_app/i18n/translations.g.dart';
+import 'package:gruene_app/swagger_generated_code/gruene_api.swagger.dart';
+
+class ProfileImageDeleter extends StatefulWidget {
+  final Profile profile;
+  final ValueChanged<Profile> onProfileUpdated;
+  final ValueChanged<bool> onProcessing;
+
+  const ProfileImageDeleter({
+    super.key,
+    required this.profile,
+    required this.onProfileUpdated,
+    required this.onProcessing,
+  });
+
+  @override
+  State<ProfileImageDeleter> createState() => _ProfileImageDeleterState();
+}
+
+class _ProfileImageDeleterState extends State<ProfileImageDeleter> {
+  Future<void> _deleteProfileImage() async {
+    final BuildContext context = this.context;
+    final theme = Theme.of(context);
+    final bool? confirmed = await showDialog<bool>(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: Text(t.profiles.profileImage.confirmDelete.title),
+          titleTextStyle: theme.textTheme.titleMedium?.apply(color: ThemeColors.textDark),
+          content: Text(t.profiles.profileImage.confirmDelete.text),
+          contentTextStyle: theme.textTheme.bodyMedium?.apply(color: ThemeColors.textDark),
+          actions: [
+            TextButton(
+              onPressed: () => Navigator.of(context).pop(false),
+              child: Text(
+                t.common.actions.cancel,
+                style: theme.textTheme.labelLarge?.apply(color: ThemeColors.textCancel),
+              ),
+            ),
+            TextButton(
+              onPressed: () => Navigator.of(context).pop(true),
+              child: Text(
+                t.common.actions.delete,
+                style: theme.textTheme.labelLarge?.apply(color: ThemeColors.textWarning),
+              ),
+            ),
+          ],
+        );
```

Comment:
:upside_down_face: Could we perhaps extract this to an own widget (same file is fine) to keep this more readable?